core/common/src/test/scala/net/liftweb/common/BoxSpec.scala

```diff
@@ -141,6 +141,15 @@ class BoxSpec extends Specification with ScalaCheck with BoxGenerator {
     "define a 'flatMap' method transforming its value in another Box. If the value is transformed in an Empty can, the total result is an Empty can" in {
       Full(0) flatMap { x: Int => if (x > 0) Full("full") else Empty } must beEmpty
     }
+    "define a 'flatten' method if it contains another Box. If the inner box is a Full can, the final result is a Full can" in {
+      Full(Full(1)).flatten must_== Full(1)
+    }
+    "define a 'flatten' method if it contains another Box. If the inner box is a Failure, the final result is a Failure" in {
+      Full(Failure("error", Empty, Empty)).flatten must_== Failure("error", Empty, Empty)
```

Comment:
@Shadowfiend IMO, In the case of Box/Option, it does make more sense to get the inner instance as that's the Box/Option with the final result (if it exists). And the actual type doesn't change, since we are getting a generic `Box` back.